tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts

```diff
@@ -171,29 +171,57 @@ export function getReleaseTag(apiItem: ApiItem): ReleaseTag | undefined {
 }
 
 /**
- * Creates a string representation of the provided release tag.
+ * Gets all {@link https://tsdoc.org/pages/spec/tag_kinds/#modifier-tags | modifier tags} associated with the provided API item.
  *
- * @remarks If `None`, this will return an empty string.
+ * @remarks Note that this will include both standard and any preserved custom tags.
+ *
+ * @public
  */
-export function releaseTagToString(releaseTag: ReleaseTag): string {
-	// eslint-disable-next-line default-case
-	switch (releaseTag) {
-		case ReleaseTag.Alpha: {
-			return "Alpha";
-		}
-		case ReleaseTag.Beta: {
-			return "Beta";
-		}
-		case ReleaseTag.Internal: {
-			return "Internal";
-		}
-		case ReleaseTag.Public: {
-			return "Public";
+export function getModifierTags(apiItem: ApiItem): ReadonlySet<string> {
+	const modifierTags = new Set<string>();
+	if (apiItem instanceof ApiDocumentedItem && apiItem.tsdocComment !== undefined) {
+		for (const tag of apiItem.tsdocComment.modifierTagSet.nodes) {
+			modifierTags.add(tag.tagName);
 		}
-		case ReleaseTag.None: {
-			return "";
+	}
+	return modifierTags;
+}
+
+/**
+ * Checks if the provided API item is tagged with the specified {@link https://tsdoc.org/pages/spec/tag_kinds/#modifier-tags | modifier tag}.
+ *
+ * @throws If the provided tag name is malformed (i.e., does not begin with `@`).
+ *
+ * @public
+ */
+export function hasModifierTag(apiItem: ApiItem, tagName: string): boolean {
```

Comment:
Might be hard depending on use cases, but there is a type for strings starting with `@`. It is:
```
`@${string}`
```

Reply:
Maybe: 
```suggestion
export function hasModifierTag(apiItem: ApiItem, tagName: string): asserts tagName is `@${string}` {
```

Reply:
This isn't a type filter. The assert isn't appropriate. (It would be nice to say `boolean & asserts ...` but Typescript doesn't support that complexity.)

Reply:
This is specific to [modifier tags](https://tsdoc.org/pages/spec/tag_kinds/#modifier-tags) which do not allow for the kinds of syntax you specified. The assert is appropriate for this function.

Reply:
If we add a function checking for inline tags, that would need to be more permissive, but we don't have a use case for that currently, to my knowledge.

Reply:
I think Craig was suggesting that adding such a type-narrowing helper might alleviate the painful cases of using a function like this with stricter type requirements.

It's an interesting suggestion, and I think I like it. I'll play around with it and see how well it works. I may not wait to merge the PR for it though.

Reply:
No, after toying with it, I don't think the effort is worth it. The `TSDoc` library uses strings to represent its tags, and we interop with their utilities a fair amount. This would require adding type-checks to assert that tags _they_ give us are valid, which seems silly.

Reply:
I did update the code to leverage TSDoc's existing validation function, though. And updated some surrounding documentation.